### DISCUSSION TITLE
Automerge is disabled if matching PR already automerged before

### DISCUSSION THREAD

**Discussion by Kuyoh** [renovatebot/renovate#22770] [Request Help] [answered]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.117.3

### If you're self-hosting Renovate, select which platform you are using.

Azure DevOps (dev.azure.com)

### Was this something which used to work for you, and then stopped?

It used to work, and then stopped

### Describe the problem

We've used to rely on automerge, which suddenly stopped working in certain PRs.
It is definitely enabled by configuration, but seems to get disabled because there is a completed PR with the same name (see logs).

The issue may be related to fixes for #22730.
The key log message seems to be:
`DEBUG: Disabling automerge because PR was merged previously (repository=dummy-repo, branch=renovate/all-minor-patch)`

I am planning to create a minimal reproducible example, but wanted to create the discussion first, in order to discuss whether it's intended behavior.

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Found closed PR with current title (repository=dummy-repo, branch=renovate/all-minor-patch)
DEBUG: getPr(31284) (repository=dummy-repo, branch=renovate/all-minor-patch)
DEBUG: getPrList() (repository=dummy-repo, branch=renovate/all-minor-patch)
DEBUG: Matching PR #31284 was merged previously (repository=dummy-repo, branch=renovate/all-minor-patch)
DEBUG: Disabling automerge because PR was merged previously (repository=dummy-repo, branch=renovate/all-minor-patch)
DEBUG: Checking if PR has been edited (repository=dummy-repo, branch=renovate/all-minor-patch)
 INFO: PR updated (repository=dummy-repo, branch=renovate/all-minor-patch)
       "pr": 31356,
       "prTitle": "Update all non-major dependencies"
```

</details>

### Have you created a minimal reproduction repository?

I have explained in the description why a minimal reproduction is impossible

**Comment by rarkins** [comment]
This is intentional behavior. Consider this scenario:

1. Renovate automerges an update
2. You discover the update has a problems, you revert it
3. Renovate repeats step 1 again

For repeat updates, you'll need to manually enable automerge

**Reply by Kuyoh** [reply]
@rarkins I guess that's true depending on the repository configuration.
It's not an issue if there is a requirement to have extra reviewers.

We are grouping multiple minor updates into a single PR using [groupname](https://docs.renovatebot.com/configuration-options/#groupname) and require extra reviewers (so it does not complete fully-automatically anyway).

In this setup, the automerge basically stops working entirely.
I'm not sure how we can achieve the behavior we need now.

Is there a way to mess with groupName/branchName in a way to make it work again?

**Reply by rarkins** [reply]
I'm not sure if there's any way today. Essentially this setting is a safety feature to ensure you can stop/roll back a bad update

**Reply by Kuyoh** [reply]
I understand that.
It is still making the automerge feature unusable when grouping PR's (with other safeguards in-place).

Would it make sense to add a new flag, something like 'ignorePreviousPrs', akin to 'ignoreTests' ?

**Reply by rarkins** [reply]
I'll take a look if we can partially revert the change to maintain previous behavior when it was a group

**Reply by patst** [reply]
we stumbled across this as well. Would be great to restore the previous behaviour (with a flag).

Otherwise the automerge feature is unusable for us as well :-/

**Reply by rarkins** [reply, marked as answer]
Should be fixed in latest release. This will be put live in the hosted app within days
